src/docfx/build/zonepivot/ZonePivotProvider.cs

```diff
@@ -0,0 +1,141 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Microsoft.Docs.Build
+{
+    internal class ZonePivotProvider
+    {
+        private readonly ErrorBuilder _errors;
+        private readonly MetadataProvider _metadataProvider;
+        private readonly FileResolver _fileResolver;
+        private readonly LinkResolver _linkResolver;
+
+        private readonly ConcurrentDictionary<FilePath, FilePath?> _zonePivotDefinitionFileCache = new ConcurrentDictionary<FilePath, FilePath?>();
+        private readonly ConcurrentDictionary<FilePath, ZonePivotGroupDefinitionModel?> _zonePivotDefinitionModelCache =
+            new ConcurrentDictionary<FilePath, ZonePivotGroupDefinitionModel?>();
+
+        public ZonePivotProvider(
+            ErrorBuilder errors,
+            MetadataProvider metadataProvider,
+            FileResolver fileResolver,
```

Comment:
`FileResolver` is used to resolve non-content files (typically configuration). Use `Input` to read content files.